spec.html

```diff
@@ -26901,89 +26901,96 @@ <h1>Overview of Date Objects and Definitions of Abstract Operations</h1>
         <p>The time value of the start of a year is:</p>
         <emu-eqn id="eqn-TimeFromYear" aoid="TimeFromYear">TimeFromYear(_y_) = msPerDay &times; DayFromYear(_y_)</emu-eqn>
         <p>A time value determines a year by:</p>
         <emu-eqn id="eqn-YearFromTime" aoid="YearFromTime">YearFromTime(_t_) = the largest integer _y_ (closest to positive infinity) such that TimeFromYear(_y_) &le; _t_</emu-eqn>
-        <p>The leap-year function is 1 for a time within a leap year and otherwise is zero:</p>
-        <emu-eqn id="eqn-InLeapYear" aoid="InLeapYear">InLeapYear(_t_)
-          = 0 if DaysInYear(YearFromTime(_t_)) = 365
-          = 1 if DaysInYear(YearFromTime(_t_)) = 366
-        </emu-eqn>
+        <p>The leap-year function is 1 for a time value within a leap year and otherwise is zero:</p>
+        <emu-eqn id="eqn-InLeapYear" aoid="InLeapYear">InLeapYear(_t_) = DayFromYear(YearFromTime(_t_) + 1) - DayFromYear(YearFromTime(_t_)) - 365</emu-eqn>
       </emu-clause>
 
       <emu-clause id="sec-month-number">
         <h1>Month Number</h1>
-        <p>Months are identified by an integer in the range 0 to 11, inclusive. The mapping MonthFromTime(_t_) from a time value _t_ to a month number is defined by:</p>
-        <emu-eqn id="eqn-MonthFromTime" aoid="MonthFromTime">MonthFromTime(_t_)
-          = 0 if 0 &le; DayWithinYear(_t_) &lt; 31
-          = 1 if 31 &le; DayWithinYear(_t_) &lt; 59 + InLeapYear(_t_)
-          = 2 if 59 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 90 + InLeapYear(_t_)
-          = 3 if 90 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 120 + InLeapYear(_t_)
-          = 4 if 120 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 151 + InLeapYear(_t_)
-          = 5 if 151 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 181 + InLeapYear(_t_)
-          = 6 if 181 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 212 + InLeapYear(_t_)
-          = 7 if 212 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 243 + InLeapYear(_t_)
-          = 8 if 243 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 273 + InLeapYear(_t_)
-          = 9 if 273 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 304 + InLeapYear(_t_)
-          = 10 if 304 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 334 + InLeapYear(_t_)
-          = 11 if 334 + InLeapYear(_t_) &le; DayWithinYear(_t_) &lt; 365 + InLeapYear(_t_)
+        <p>Months are identified by an integer in the range 0 to 11, inclusive. A month value of 0 specifies January; 1 specifies February; 2 specifies March; 3 specifies April; 4 specifies May; 5 specifies June; 6 specifies July; 7 specifies August; 8 specifies September; 9 specifies October; 10 specifies November; and 11 specifies December.</p>
+        <p>The position of a month _m_ within a year depends upon whether or not the year is a leap year:</p>
+        <emu-eqn id="eqn-DaysWithinYearBeforeMonth" aoid="DaysWithinYearBeforeMonth">DaysWithinYearBeforeMonth(_m_, _inLeapYear_)
```

Comment:
it's strange to have a variable named "in leap year", which implies it's a boolean, but use it in arithmetic. any reason this couldn't be a boolean, and the addition here done with `ToNumber()`?

Reply:
This stems from the [definition of the operation](https://tc39.es/ecma262/#eqn-InLeapYear):
> The leap-year function is 1 for a time within a leap year and otherwise is zero

Would you consider it an improvement to change its argument from a time value to a year and update its name to e.g. LeapDaysInYear, supporting changes like this?
```diff
- 1. Let _inLeapYear_ be InLeapYear(_t_).
+ 1. Let _leapDaysInYear_ be leapDaysInYear(YearFromTime(_t_)).
```

Reply:
that would certainly seem clearer to me :-)

Reply:
Done.